File: JULY-2025/10-07-2025/p3440.java
class Solution {
    public int maxFreeTime(int eventTime, int[] startTime, int[] endTime) {
        int n = startTime.length;
        if (n == 1) {
            return eventTime - (endTime[0] - startTime[0]);
        }
        int[] gaps = new int[n + 1];
        gaps[0] = startTime[0];
        for (int i = 1; i < n; i++) {
            gaps[i] = startTime[i] - endTime[i - 1];
        }
        gaps[n] = eventTime - endTime[n - 1];
        int[] maxLeft = new int[n + 1];
        int[] maxRight = new int[n + 1];
        maxLeft[0] = gaps[0];
        for (int i = 1; i < n; i++) {
            maxLeft[i] = Math.max(maxLeft[i - 1], gaps[i]);
        }
        maxRight[n] = 0;
        for (int i = n - 1; i >= 0; i--) {
            maxRight[i] = Math.max(maxRight[i + 1], gaps[i + 1]);
        }
        int maxFree = 0;
        int firstDuration = endTime[0] - startTime[0];
        if (maxRight[1] >= firstDuration) {
            maxFree = Math.max(maxFree, gaps[0] + gaps[1] + firstDuration);
        }
        maxFree = Math.max(maxFree, gaps[0] + gaps[1]);
        for (int i = 1; i < n; i++) {
            int duration = endTime[i] - startTime[i];
            if (maxLeft[i - 1] >= duration || maxRight[i + 1] >= duration) {
                maxFree = Math.max(maxFree, gaps[i] + gaps[i + 1] + duration);
            }
            maxFree = Math.max(maxFree, gaps[i] + gaps[i + 1]);
        }
        return maxFree;
    }
}

File: JULY-2025/10-07-2025/p3440.php
class Solution {

    /**
     * @param Integer $eventTime
     * @param Integer[] $startTime
     * @param Integer[] $endTime
     * @return Integer
     */
    function maxFreeTime($eventTime, $startTime, $endTime) {
        $n = count($startTime);
        // Initial maximum free time before first meeting and after last meeting
        $maxBeforeFirst = $startTime[0];
        $maxAfterLast = $eventTime - $endTime[$n - 1];
        
        // Calculate initial maximum considering adjacent meetings
        $maxFree = max(
            $maxBeforeFirst + ($n > 1 ? $startTime[1] - $endTime[0] : 0),
            $maxAfterLast + ($n > 1 ? $startTime[$n - 1] - $endTime[$n - 2] : 0)
        );
        
        // Initialize variables to track maximum gaps before and after each meeting
        $maxGapBefore = $maxBeforeFirst;
        $maxGapAfter = $maxAfterLast;
        for ($i = 1; $i < $n; $i++) {
            $k = $n - 1 - $i; // Mirror index for the end of the array
            
            // Calculate potential free time by moving current meeting
            // For meetings in the first half of the array
            $nextStart = ($i < $n - 1) ? $startTime[$i + 1] : $eventTime;
            $currentDuration = $endTime[$i] - $startTime[$i];
            $potentialFree = $nextStart - $endTime[$i - 1] - ($currentDuration <= $maxGapBefore ? 0 : $currentDuration);
            $maxFree = max($maxFree, $potentialFree);
            
            // For meetings in the second half of the array
            $prevEnd = ($k > 0) ? $endTime[$k - 1] : 0;
            $currentDurationK = $endTime[$k] - $startTime[$k];
            $potentialFreeK = $startTime[$k + 1] - $prevEnd - ($currentDurationK <= $maxGapAfter ? 0 : $currentDurationK);
            $maxFree = max($maxFree, $potentialFreeK);
            
            // Update maximum gaps
            $maxGapBefore = max($maxGapBefore, $startTime[$i] - $endTime[$i - 1]);
            $maxGapAfter = max($maxGapAfter, ($k < $n - 1 ? $startTime[$k + 1] - $endTime[$k] : 0));
        }
        
        return $maxFree;
    }
}